proto_parser/src/proto_parser.h

```diff
@@ -30,10 +30,20 @@ class ProtoParser {
 
   ProtoParser() {}
 
-  ProtoParser(absl::string_view text_proto);
+  ProtoParser(absl::string_view text_proto) {
+    lines_ = absl::StrSplit(text_proto, '\n');
+  }
+
+  struct PrevFieldLine {
+
+    PrevFieldLine(int line_in) 
+      : line(line_in) {}
+
+    int line;
+  };
 
   // Populates the message with all the fields of the message.
-  void PopulateFields(int& last_field_loc, 
+  void PopulateFields(PrevFieldLine& last_field_loc, 
```

Comment:
As I mentioned lots of times, if you are about to change value for a input param, use PrevFieldLine*; if  you are just reading the value, use "const PrevFieldLine&".

Recently PrevFieldLine& is becoming allowed, iff you can give it a readable name that let reader know the function may change the value inside.

Reply:
My bad my thought was since prevfieldline consists only a int which is 4 byte it's more efficient to pass reference to that than an 8 byte pointer, but I forgot about the style guide they are changed to pointers now.